File: fizzbuzz.kt
// Fizzbuzz in Kotlin
//
// To run : 
//   $ kotlinc fizzbuzz.kt -include-runtime -d fizzbuzz.jar
//   $ java -jar fizzbuzz.jar
//
// Christopher Brunsdon
// christopher@brunsdon.co.za
// 

fun fizzbuzz() {
  for (i in 1..100) {
    var result:String = "";
    if (i%3 == 0) {
      result += "fizz";
    }
    if (i%5 == 0) {
      result += "buzz";
    }
    if(result == "") {
      result = i.toString();
    }
    println(result);
  }
}

fun main(args: Array<String>) {
  fizzbuzz();
}

File: fizzbuzz.rb
# Christopher Brunsdon
# christopher@brunsdon.co.za
# Ruby Fizzbuzz

def fizzbuzz(i)
  result = %Q()
  result << 'fizz' if i.remainder(3).zero?
  result << 'buzz' if i.remainder(5).zero?
  result.empty? ? i.to_s : result
end

1.upto(100) do |n|
  puts fizzbuzz(n)
end
